### PR TITLE
Guard compiler defines in fmt

### LINKED ISSUE #802: Axom's fmt still bleeds into applications
If I include fmt/core.h and axom/fmt/fmt.hpp (the latter through axom/quest.hpp), I run into compilation errors. Basically, whichever is included first wins, and since the namespaces used are different, the code will always fail to compile.

```
In file included from axom/include/axom/fmt.hpp:15:
axom/include/axom/fmt/chrono.h:312:23: error: no member named 'fmt' in namespace 'axom'; did you mean simply 'fmt'?
      using namespace axom::fmt::detail;
                      ^

axom/include/axom/fmt/chrono.h:20:1: note: 'fmt' declared here
FMT_BEGIN_NAMESPACE
^
fmt/include/fmt/core.h:238:15: note: expanded from macro 'FMT_BEGIN_NAMESPACE'
    namespace fmt {           \
              ^
```

I doubt this is the best solution, but the only foolproof way I can think of to fix this right now is to prefix all macros in Axom's fmt with AXOM_.

### PR DESCRIPTION
**Pull request by white238**
Fixes #802 

@adayton1

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]
Looks good to me. I did not check every single instance since I'm assuming you used sed.

**Comment by white238** [comment]
> Looks good to me. I did not check every single instance since I'm assuming you used sed.

I did a search and replace on every file.  I initially walked through several files to make sure I wasn't missing some case then blanket did every file.

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

Since we'll need to reapply this whenever we update `fmt` should we have a script checked in to apply these (perhaps in a follow-up PR) ?

In either case, please document somewhere that this change needs to happen when we update `fmt`.  Some possibilities:
* `fmt.hpp` 
* `thirdparty/CMakeLists.txt`
* somewhere in the sphinx dev guide, e.g. https://axom.readthedocs.io/en/develop/docs/sphinx/dev_guide/updating_tpls.html#updating-tpls

**Comment by white238** [comment]
> Thanks @white238 !
> 
> Since we'll need to reapply this whenever we update `fmt` should we have a script checked in to apply these (perhaps in a follow-up PR) ?
> 
> In either case, please document somewhere that this change needs to happen when we update `fmt`. Some possibilities:
> 
>     * `fmt.hpp`
> 
>     * `thirdparty/CMakeLists.txt`
> 
>     * somewhere in the sphinx dev guide, e.g. https://axom.readthedocs.io/en/develop/docs/sphinx/dev_guide/updating_tpls.html#updating-tpls

Done.

File: src/thirdparty/CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,14 @@ mark_as_advanced(CLI11_FOUND)
 # Register the built-in fmt library for string formatting
 #------------------------------------------------------------------------------
 
+# NOTE: When updating fmt you must do the following changes to properly guard
+#  our fmt from other fmt's.
+# 
+# 1) Change the namespace from 'fmt' to 'axom::fmt'.  This is in fmt/core.h:203-204,
+#    as well as, search and replace `fmt::` to `axom::fmt`.
+# 2) Search and replace compiler define's prefix from `FMT_` to `AXOM_FMT_`.
+# 3) Keep fmt's headers installed under axom's include directory.
```
Comment (kennyweiss) [inline src/thirdparty/CMakeLists.txt:66]: Thanks @white238 !

**Review by adayton1 — APPROVED** [review]
LGTM

**Review by rhornung67 — APPROVED** [review]
Thanks @white238 for working through this.

**Review by agcapps — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**